### PR TITLE
PCOM-6286 Minimum Date Validation Failed

### PR DESCRIPTION
**Pull request by chadhav-ae**
### Issues Fixed
Minimum Date Validation Failed on same day.

### CODE REVIEW

**Review by kdelorey — APPROVED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export class CustomDateValidator {
                 const inputDate = new Date(field.value);
                 if (!isNaN(inputDate.getTime())) {
                     const minDate = this._resolveTime(minTime);
+                    minDate.setDate(minDate.getDate() - 1);
```
Comment (kdelorey) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:12]: This should be done at the server not here. The server sets the minDate as the current date and it needs to be "- 1". This change here affects all date validation now.

**Review by kdelorey — CHANGES_REQUESTED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts (hunk 1 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:12]: Change needs to happen in `TenderShoppingPassState.java` line `220`

Comment (chadhav-ae) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:12]: But does we know all date are not failing on same day. I updated the server first than found all date should be failing this check if we use min date as same date.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export class CustomDateValidator {
...
                     if (inputDate < minDate) {
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:13]: Couldn't you just do this?
```suggestion
                    if (inputDate <= minDate) {
```

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts (hunk 1 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:12]: You should also keep in mind that `Date` objects also include `Time`. I think what you really want to do is construct a date of "Today" at 00:00:00

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts (hunk 2 above)
Comment (chadhav-ae) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:13]: This is not working as well as inputDate is having a time and min Date is having time which is different. So the date comparing is showing a different result.

Comment (jchirico8) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/validators/custom-date-validators.ts:13]: That makes sense. I think Kevin's idea to set the time on min date would work best

**Pull request closed without merging**